core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceInfo.java

```diff
@@ -60,6 +68,15 @@ void setAdTrackingEnabled(boolean enabled) {
 
     void setAdvertisingId(String advertisingId) {
         this.advertisingId = advertisingId;
+        this.adTrackingEnabled = (this.advertisingId != null);
+        if (RudderClient.getApplication() != null) {
+            RudderPreferenceManager preferenceManager = RudderPreferenceManager.getInstance(RudderClient.getApplication());
+            preferenceManager.saveAdvertisingId(advertisingId);
```

Comment:
As we get the `preferenceManager` object as an argument on the constructor, let's utilise that.

Reply:
agreed, just pushed the changes for the same.